Bottom: e50f65b717e92423e8fb8deb12176c88c0b87ff6
Top:    60ee0e8445bbf88bbd2d4588e41ef5bbff682553
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 15:29:01 +0000

Try not using no-holes dicer for drawing polygons











---

diff --git a/src/draw.c b/src/draw.c
index 7a506d0..6eb3823 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2224,6 +2224,7 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
     }
   else if (Polygon->Clipped)
     {
+#if 0
       if (!Polygon->NoHolesValid)
         {
           ComputeNoHoles (Polygon);
@@ -2237,6 +2238,15 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
             poly.Clipped = poly.Clipped->f;
           } while (poly.Clipped != Polygon->NoHoles);
         }
+#endif
+        {
+          PolygonType poly = *Polygon;
+          poly.Clipped = Polygon->Clipped;
+          do {
+            DrawPolygonLowLevel (&poly, NULL);
+            poly.Clipped = poly.Clipped->f;
+          } while (poly.Clipped != Polygon->Clipped);
+        }
       /* draw other parts of the polygon if fullpoly flag is set */
       /* NB: No "NoHoles" cache for these */
       if (TEST_FLAG (FULLPOLYFLAG, Polygon))
